### PR TITLE
DOCSP-39814: Retrieve guide

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-40798
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-39814-retrieve/read/retrieve/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read/retrieve.txt [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+and assign the following values to your ``db`` and ``collection`` variables:
+
+.. literalinclude:: /includes/read/retrieve.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-db-coll
+    :end-before: end-db-coll
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<cpp-get-started>`.
```
Comment (norareidy) [inline source/read/retrieve.txt:43]: Note to reviewer: this link will be broken until #13 is merged

File: source/read/retrieve.txt [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+   * - Field
+     - Description
+
+   * - ``batch_size`` 
+     - | The number of documents to return per batch.
+       | Type: ``std::int32_t``
+
+   * - ``collation`` 
+     - | The collation to use for this operation.
```
Comment (jordan-smith721) [inline source/read/retrieve.txt:182]:
```suggestion
     - | The collation to use for the operation.
```

File: source/read/retrieve.txt [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Field
+     - Description
+
+   * - ``batch_size`` 
+     - | The number of documents to return per batch.
+       | Type: ``std::int32_t``
```
Comment (jordan-smith721) [inline source/read/retrieve.txt:179]: Q: Should we bold `Type:` here? No strong opinions, but just a thought

**Review by jordan-smith721 — APPROVED** [review]
LGTM w/ two small things!

File: source/read/retrieve.txt (hunk 3 above)
Comment (norareidy) [reply source/read/retrieve.txt:179]: Yeah I like that idea!

File: source/includes/read/retrieve.cpp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    // end-find-one
+
+    // start-find-many
+    auto cursor = collection.find(make_document(kvp("founded_year", 1970)));
+    // end-find-many
+
+    // start-cursor
+    auto cursor = collection.find(make_document(kvp("founded_year", 1970)));
+    for(auto&& doc : cursor) {
+        std::cout << bsoncxx::to_json(doc) << "\n" << std::endl;
```
Comment (kevinAlbs) [inline source/includes/read/retrieve.cpp:37]:
```suggestion
        std::cout << bsoncxx::to_json(doc) << std::endl;
```

Remove duplicate newline: `\n` and `std::endl`.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**